src/gui/folderwatcher.cpp

```diff
@@ -190,4 +206,62 @@ void FolderWatcher::changeDetected(const QStringList &paths)
     }
 }
 
+void FolderWatcher::possiblyAddUnlockedOfficeFile(const QString &path, QSet<QString> &container)
+{
+    qCInfo(lcFolderWatcher) << "Checking if it is an office lock file:" << path;
+    QString officeLockFilePrefixFound;
+    for (const auto &currentPrefix : officeLockPrefixes) {
+        if (path.contains(currentPrefix)) {
```

Comment:
@mgallien path will never start with prefix, it is an absolute path and we are looking for a prefix in file's name, e.g. `C:/users/whatever/Nextcloud/.~lock.file.docx#` - this will always fail if I use `startsWith` and it is a prefix but for a lock file, not for absolute path

Reply:
@mgallien regular expression will not always work, for example in case we also have a postfix (`#` in case of LibreOffice), and (first 2 characters are removed from the file name in case of `Office365` so, e.g. `document.docx` becomes `~$cument.docx`), so parsing is required and we can't just get away with using regular expressions (besides, I am not a fan of using RegExp in cases)